doc/dev/crimson/poseidonstore.rst

```diff
@@ -0,0 +1,551 @@
+is used to map any collections to SPs.
+We can use BlueStore's approach for collection splitting, changing the number of significant bits for the collection prefixes.
+Because the prefixes of the parent collection ID do not change even after collection splitting, the mapping between 
+the collection and SP are maintained.
+The number of SPs may be configured to match the number of CPUs allocated for each disk so that each SP can hold 
+a number of objects large enough for cross-SP transaction not to occur.
+
+In case of need of cross-SP transaction, we could use the global WAL. The coordinator thread (mainly manages global partition) handles 
+cross-SP transaction via acquire the source SP and target SP locks before processing the cross-SP transaction. 
+Source and target probably are blocked.
+
+For the load unbalanced situation, 
+Poseidonstore can create partitions to make full use of entire space efficiently and provide load balaning.
+
+
+CoW/Clone
+---------
+As for CoW/Clone, a clone has its own onode like other normal objects.
+
+Although each clone has its own onode, data blocks should be shared between the original object and clones
+if there are no changes on them to minimize the space overhead.
+To do so, the reference count for the data blocks is needed to manage those shared data blocks.
+
+To deal with the data blocks which has the reference count, poseidon store makes use of shared_blob
+which maintains the referenced data block.
+
+As shown the figure as below, 
+the shared_blob tracks the data blocks shared between other onodes by using a reference count.
+The shared_blobs are managed by shared_blob_list in the superblock.
+
+
+.. ditaa::
+
+                          
+    +----------+         +----------+
+    | Object A |         | Object B |
+    +==========+         +==========+
+    | Extent   |         | Extent   |
+    +----------+         +----------+
+        |  |     +----------|    |
+        |  +---------------+     |
+        |        |         |     |
+        v        v         v     v
+    +---------------+---------------+
+    | Data block 1  | Data block 2  |
+    +---------------+---------------+ 
+            |              |
+            v              v 
+    +---------------+---------------+
+    | shared_blob 1 | shared_blob 2 |
+    +===============+---------------+ shared_blob_list
+    | refcount      | refcount      |
+    +---------------+---------------+
+
+
+
+Plans
+=====
+
+All PRs should contain unit tests to verify its minimal functionality.
+
+* WAL and block cache implementation
+
+  As a first step, we are going to build the WAL including the I/O procedure to read/write the WAL.
+  With WAL development, the block cache needs to be developed together.
+  Besides, we are going to add an I/O library to read/write from/to the NVMe storage to 
+  utilize NVMe feature and the asynchronous interface.
+
+* Radix tree and onode 
```

Comment:
There is confusion above about what a radix tree is and how you mean to use it.  For now, add a planning item here:

```First, submit a PR against this file with a more detailed on disk layout and lookup strategy for the onode radix tree.

Follow up with implementations based on the above design once design PR is merged.```

I'm also not sure what you mean below by "onode can interact between onodes".

Reply:
Updated.